server/api/audit_boards.py

```diff
@@ -260,41 +260,49 @@ def calculate_risk_measurements(election: Election, round: Round):
         raise Exception("Risk limit not defined")
     risk_limit: int = election.risk_limit
 
-    for contest in election.contests:
+    for round_contest in round.round_contests:
+        contest = round_contest.contest
         risk, is_complete = bravo.compute_risk(
             float(risk_limit) / 100,
             sampler_contest.from_db_contest(contest),
             cumulative_contest_results(contest),
         )
 
-        round_contest = next(
-            rc for rc in round.round_contests if rc.contest_id == contest.id
-        )
         round_contest.end_p_value = max(risk.values())
         round_contest.is_complete = is_complete
 
 
 def count_audited_votes(round: Round):
     for round_contest in round.round_contests:
         contest = round_contest.contest
-        vote_counts = dict(
-            BallotInterpretation.query.filter_by(
-                contest_id=contest.id, is_overvote=False
+        # For a targeted contest, count the ballot draws sampled for the contest
+        if contest.is_targeted:
+            vote_counts = dict(
+                BallotInterpretation.query.filter_by(
+                    contest_id=contest.id, is_overvote=False
+                )
+                .join(SampledBallot)
+                .join(SampledBallotDraw)
+                .filter_by(round_id=round.id, contest_id=contest.id)
+                .join(BallotInterpretation.selected_choices)
+                .group_by(ContestChoice.id)
+                .values(ContestChoice.id, func.count(SampledBallot.id))
             )
-            .join(SampledBallot)
-            .join(SampledBallotDraw)
-            .filter_by(round_id=round.id)
-            .join(BallotInterpretation.selected_choices)
-            .group_by(ContestChoice.id)
-            .values(
-                ContestChoice.id,
-                func.count(
-                    SampledBallot.id
-                    if contest.is_targeted
-                    else SampledBallot.id.distinct()
-                ),
+        # For an opportunistic contest, count the unique ballots that were
+        # audited for this contest, regardless of which contest they were
+        # sampled for.
+        else:
+            vote_counts = dict(
+                BallotInterpretation.query.filter_by(
+                    contest_id=contest.id, is_overvote=False
+                )
+                .join(SampledBallot)
+                .join(SampledBallotDraw)
+                .filter_by(round_id=round.id)
+                .join(BallotInterpretation.selected_choices)
+                .group_by(ContestChoice.id)
+                .values(ContestChoice.id, func.count(SampledBallot.id.distinct()))
```

Comment:
I had tried that before and it didn't work. Just tried it again to remember why: that query returns a count of 1 for each ContestChoice.id. If you can help me think through why that is, I'd welcome the help, because I didn't and still don't understand why it's happening

Reply:
I had tried that before and it didn't work. Just tried it again to remember why: that query returns a count of 1 for each ContestChoice.id. If you can help me think through why that is, I'd welcome the help, because I didn't and still don't understand why it's happening

Reply:
if you've got it working with your query, I'm good with it. Worth a longer debug on the other one at some point, but don't want to block you today.